File: backend/app/core/upgrades/curses/damage_reduc_5.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class DamageReduc5(UpgradeImplementation):
    id = 'damage_reduc_5'
    name = 'Weakness 5'
    description = '-3.75% damage.'
    rarity = 'legendary'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -0.0375
        appliers.modifiers.add_multiplier('attack', 'damage', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/curses/damage_reduc_5.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const DamageReduc5Def: UpgradeDef = {
  id: "damage_reduc_5",
  name: "Weakness 5",
  description: "-3.75% damage.",
  rarity: RarityID.Legendary,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Attack,
  fieldInTargetClass: UpgradeStatID.Damage,
  value: -0.0375,
  isMultiplier: true,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class DamageReduc5 extends Upgrade {}
